Classes/Domain/Repository/LanguageSkillRepository.php

```diff
@@ -27,15 +27,13 @@ class LanguageSkillRepository extends Repository
     public function findAllWithEmployeeRelation(): QueryResultInterface
     {
         $query = $this->createQuery();
-
         $query->matching(
             $query->logicalAnd(
-                [
                 $query->equals('employee.hidden', 0),
-                $query->equals('employee.deleted', 0)
-                ]
-            )
+                $query->equals('employee.deleted', 0),
```

Comment:
I don't understand that query as deleted=0 and hidden=0 is default. As this is a relation we have to look here: 

https://github.com/TYPO3/typo3/blob/v11.5.39/typo3/sysext/extbase/Classes/Persistence/Generic/Mapper/DataMapper.php#L389C9-L389C53

The normal QueryFactory will be called with default QuerySettings. And default query settings is deleted=0 and hidden=0.

Something like `employee.uid IS NOT NULL` should happen here

Reply:
Not sure. By doing this it ensures that the property employee is set. You got my point??